File: src/lib/types/epg.ts
export interface EpgDisplayName {
    lang: string;
    value: string;
}

export interface EpgIcon {
    src: string;
    width?: number;
    height?: number;
}

export interface EpgChannel {
    id: string;
    displayName: EpgDisplayName[];
    icon: EpgIcon[];
    url: string[];
}

export interface EpgProgram {
    id: string;
    channelId: string;
    start: string;
    stop: string;
    title: string;
    description?: string;
    category?: string;
    icon?: string;
}

export interface EpgItem {
    id: string;
    epgId: string;
    title: string;
    lang: string;
    start: string;
    end: string;
    stop: string;
    description: string;
    channelId: string;
    startTimestamp: string;
    stopTimestamp: string;
}

File: src-tauri/src/types/epg.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct EpgProgram {
    pub id: String,
    pub channel_id: String,
    pub start: String,
    pub stop: String,
    pub title: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub category: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub icon: Option<String>,
}
